applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_law.cpp

```diff
@@ -85,8 +83,8 @@ double& GeoLinearElasticLaw::CalculateValue(ConstitutiveLaw::Parameters& rParame
                                             double&                      rValue)
 {
     if (rThisVariable == STRAIN_ENERGY) {
-        Vector& r_strain_vector = rParameterValues.GetStrainVector();
-        Vector& r_stress_vector = rParameterValues.GetStressVector();
+        const Vector& r_strain_vector = rParameterValues.GetStrainVector();
+        Vector&       r_stress_vector = rParameterValues.GetStressVector();
```

Comment:
Now that we're touching these lines, we might as well replace the explicitly repeated types by `auto`:
```suggestion
        const auto& r_strain_vector = rParameterValues.GetStrainVector();
        auto&       r_stress_vector = rParameterValues.GetStressVector();
```